lib/gvl-tracing.rb

```diff
@@ -28,6 +28,60 @@
 require_relative "gvl_tracing/version"
 
 require "gvl_tracing_native_extension"
+require "json"
 
 module GvlTracing
+  @path = "/tmp/gvl-tracing.json"
+  class << self
+    def start(file)
+      @path = file
+      GvlTracing._start(@path)
+    end
+
+    def stop
+      thread_list = Thread.list
+
+      GvlTracing._stop
+
+      set_thread_name(thread_list)
+    end
+
+    private
+
+    def set_thread_name(list)
+      threads = aggreate_thread_list(list)
+      output_file = File.open(@path)
+      output = output_file.read.split("\n").map do |event|
+        parse_event = JSON.parse(event)
+        thread_id = parse_event.dig("args", "name")
+
+        next(parse_event) unless thread_id
+
+        label = threads[thread_id.strip.to_i]
+
+        next(parse_event) unless label
+
+        parse_event["args"]["name"] = label
+        parse_event
+      end
+      File.write(@path, output.to_json)
+    end
+
+    def aggreate_thread_list(list)
+      list.each_with_object({}) do |t, acc|
+        next unless t.name
+
+        acc[t.native_thread_id] = thread_label(t)
+      end
+    end
+
+    REGEX = /lib(?!.*lib)\/([a-zA-Z-]+)/
+    def thread_label(thread)
+      lib_name = thread.to_s.match(REGEX)
```

Comment:
If your goal is to exact a gem name or something, then `thread.backtrace_locations.last.path` might be cleaner.

Reply:
I look at it a little bit. But `bracktrace` is an array of strings. 
The last element looks very similar to Thread instance `to_s`
```
thread.backtrace.last
=> "/Users/benoit.tigeot/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/concurrent-ruby-1.2.2/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:333:in `block in create_worker'"
```

Reply:
I was pointing at `bracktrace_locations`, not `backtrace`.

Also to avoid allocation too much useless stuff: `t.backtrace_locations(1, 1).first.path`

Reply:
I tried an the main issue is we may do not have path.

So diff is
```diff
--- a/lib/gvl-tracing.rb
+++ b/lib/gvl-tracing.rb
@@ -81,7 +81,7 @@ module GvlTracing

     REGEX = /lib(?!.*lib)\/([a-zA-Z-]+)/
     def thread_label(thread)
-      lib_name = thread.to_s.match(REGEX)
+      lib_name = thread.backtrace_locations(1, 1).first&.path&.match(REGEX)

Reply:
Right, could be a native method.

Reply:
Do you think we should still change it to `backtrace_locations`? From outside I see limited gains.

Reply:
I don't have a strong opinion on this one (yet? xD), so let's go with what you picked for now :)